packages/gator-permissions-snap/src/clients/blockchainMetadataClient.ts

```diff
@@ -72,143 +82,174 @@ export class BlockchainTokenMetadataClient implements TokenMetadataClient {
       throw new InvalidInputError(message);
     }
 
-    // Check if we're on the correct chain
-    const selectedChain = await this.#ethereumProvider.request<Hex>({
-      method: 'eth_chainId',
-      params: [],
-    });
-
-    if (selectedChain && hexToNumber(selectedChain) !== chainId) {
-      await this.#ethereumProvider.request<Hex>({
-        method: 'wallet_switchEthereumChain',
-        params: [{ chainId: numberToHex(chainId) }],
-      });
-
-      const updatedChain = await this.#ethereumProvider.request<Hex>({
-        method: 'eth_chainId',
-        params: [],
-      });
-
-      if (updatedChain && hexToNumber(updatedChain) !== chainId) {
-        throw new ChainDisconnectedError(
-          'Selected chain does not match the requested chain',
+    const { retries = 1, delayMs = 1000 } = retryOptions ?? {};
```

Comment:
should this be leveraging the `makeValidatedRequestWithRetry` function to avoid having to implement it's own retry mechanism?